petastorm/reader_impl/worker_loop.py

```diff
@@ -49,15 +82,21 @@ def __str__(self):
 
 
 # Runs in a separate thread
+def dispatch_decode(decoder, shuffled):
+    return decoder.decode(shuffled)
+
+
 def worker_loop(epochs_generator, loading_pool, loader, decoding_pool, decoder, shuffling_queue, output_queue,
-                stop_event, stats):
+                stop_event, row_serializer, stats):
     # Possibly infinite loop driven by row-group ventilation logic (infinite if epochs=inf)
     in_loading_futures = []
     in_decoding = []
 
     if not isinstance(stats, Counter):
         raise ValueError('stats argument is expected to be a collections.Counter')
 
+    chunk_decoding_size = max(1, decoding_pool._max_workers / _OUTPUT_QUEUE_SIZE)
```

Comment:
in python3 this can yield a float if `_max_workers` > 30. That seems like a bad thing. 

Also this seems to pretty much always be 1 since it should be rare to have more than 30 threads/processes anyways so Im not sure I understand the purpose of this.

Reply:
Yes. This does not make sense. Should build some sort of a benchmark to measure performance for different datasets structures to be smarter about this. Modifed `chunk_decoding_size` to become 1/10 of the rowgroup size when it becomes known.